src/waveform/renderers/qopengl/waveformrenderbeat.cpp

```diff
@@ -68,7 +68,9 @@ void WaveformRenderBeat::renderGL() {
     const float rendererBreadth = m_waveformRenderer->getBreadth();
 
     const int numVerticesPerLine = 6; // 2 triangles
-    const int numBeatsInRange = trackBeats->numBeatsInRange(startPosition, endPosition);
+    const int numBeatsInRange = startPosition.isValid() && endPosition.isValid()
```

Comment:
I'm not sure, but I checked other uses of numBeatsInRange, and there the start and end position are also checked. I have not been able to reproduce the issue, but looking at the code it does make sense that the asserts that @uklotzde reported fails when the positions are invalid. Anyway, it certainly doesn't hurt to check!

Reply:
Mhmm. Then the thing I'm concerned about is whether 0 is actually a good default value in case of failure. Also in case its 0, the debug assert later in the code will also trip because the vector is larger than 0.

Reply:
Will do.

I could also remove the call to numBeatsInRange altogether and just reserve a fixed size that is sufficient to contain the vertices for a maximum amount of beats. That might be the safest solution for now.